pytorch_lightning/trainer/data_loading.py

```diff
@@ -201,6 +201,11 @@ def _get_dataloader_init_kwargs(
 
         # get the dataloader instance `__init__` parameters
         params = dict(inspect.signature(dataloader.__init__).parameters)
+        has_variadic_kwargs = any(p.kind is p.VAR_KEYWORD for p in params.values())
+        if has_variadic_kwargs:
+            # if the signature takes **kwargs, assume they will be passed down with `super().__init__(**kwargs)`
+            params.update(inspect.signature(DataLoader.__init__).parameters)
+            del params["self"]
 
         # keep only the params whose default is different to the current attr value
         non_defaults = {name for name, p in params.items() if name in attrs and p.default != attrs[name]}
```

Comment:
Small note: This would break if default is an array.

Reply:
Mutable objects are not supposed to be used as signature defaults